hail/src/main/scala/is/hail/expr/types/physical/PPrimitive.scala

```diff
@@ -38,4 +43,11 @@ trait PPrimitive extends PType {
     assert(srcPType.isOfType(this))
     Region.copyFrom(srcAddress, addr, byteSize)
   }
+
+  override def constructAtAddressFromValue(mb: MethodBuilder, addr: Code[Long], region: Code[Region], srcPType: PType, src: Code[_], forceDeep: Boolean): Code[Unit] = {
+    assert(this.isOfType(srcPType))
+    storePrimitiveAtAddress(addr, srcPType, src)
+  }
+
+  def storePrimitiveAtAddress(addr: Code[Long], srcPType: PType, value: Code[_]): Code[Unit]
```

Comment:
This seems more complex than what we had before (2 functions that were composed, copyFromType and storeShallowAtOffset, which has the same functionality as storePrimitiveAtAddress). 

Can this be folded in with constructAtAddressAndStackValue (this seems like it may be more inline with the goals of the reorg in the rvb pr)? Is Region.copyFrom necessary if you know the byteSize explicitly?

Reply:
This is a helper function that has a simpler signature than constructAtAddressFromValue. We could also  leave the inheritors of PPrimitive to define `constructAtAddressFromValue`, but this seems marginally simpler.

Reply:
We need these interfaces because the previous interfaces didn't cover the use case in SRVB.

Reply:
I don’t entirely see why the new function can’t be in copyFromTypeAndStackValue, behind a forceDeep flag. Approved, but would like to make sure I’m convinced of this, at some point, if you have time.